rust/src/osp/commands.rs

```diff
@@ -188,7 +188,13 @@ fn write_scanner_prefs(scan: &Scan, writer: &mut Writer) -> Result<()> {
     writer.write_event(Event::Start(BytesStart::new("scanner_params")))?;
     for p in &scan.scan_preferences {
         writer.write_event(Event::Start(BytesStart::new(&p.id)))?;
-        writer.write_event(Event::Text(BytesText::new(&p.value)))?;
+        let mut value = p.value.as_str();
```

Comment:
In rust you don't need to create a variable before assigning it. It is considered a bad practice.

A better way to handle it would be:

```
        let value = match p.value.as_ref() {
            "yes" => "1",
            "no" => "0",
            v => v,
        };
        writer.write_event(Event::Text(BytesText::new(value)))?;
```

Reply:
Thanks Philipp, I thought it would need improvement. Changed in [ed8f23e](https://github.com/greenbone/openvas-scanner/pull/1805/commits/ed8f23ea9f1a7520d661ec366242f30e2c1bbb46).